### ISSUE TITLE
Run the PHP language server exception using lsp-tcp-connection

### ISSUE THREAD

**Issue by tszg** [emacs-lsp/lsp-mode#514]
**Describe the bug**
Run the PHP language server exception using lsp-tcp-connection

**To Reproduce**
Install lsp-mode and PHP language server.
My configuration:\
```emacs-lisp
(setq lsp-clients-php-server-command
        `("php" ,(concat server-dir "php-ls/vendor/felixfbecker/language-server/bin/php-language-server.php" " --tcp-server=localhost:29543" " --memory-limit=4095M")))
  (lsp-register-client
   (make-lsp-client :new-connection (lsp-tcp-connection
                                     (-const lsp-clients-php-server-command) "localhost" 29543)
                    :major-modes '(php-mode)
                    :server-id 'php-ls))
```

**Expected behavior**
Connect and run lsp php

**Which Language Server did you use**
lsp-php lsp.el

**OS**
Windows10

**Error callstack**
Debugger entered--Lisp error: (error "make client process failed: Connection refused, :name, php-ls TCP connection, :buffer, nil, :host, localhost, :service, 29543, :nowait, nil, :tls-parameters, nil")
  signal(error ("make client process failed: Connection refused, :name, php-ls TCP connection, :buffer, nil, :host, localhost, :service, 29543, :nowait, nil, :tls-parameters, nil"))
  error("make client process failed: Connection refused, :name, php-ls TCP connection, :buffer, nil, :host, localhost, :service, 29543, :nowait, nil, :tls-parameters, nil")
  (progn (lsp--spinner-stop) (error (error-message-string err)))

**Comment by klajo** [comment]
I bumped into this as well, but with [sourcer for erlang](https://github.com/erlang/sourcer).

To me it looks like if the make-process:ed program which will be the TCP server isn't given enough time/is too slow to start before the open-network-stream is run ==> connection refused.  This is [the code I'm talking about](https://github.com/emacs-lsp/lsp-mode/blob/master/lsp.el#L2865).

I tried a quick hack and added simple delay and then it connected.

Cheers,
Klas

**Comment by yyoncho** [comment]
I will provide a free port from lsp.el but I would recommend stdio .

**Comment by klajo** [comment]
Thanks.

**Comment by tszg** [comment]
Yes, I know that. But in the windows operating system, you need to use tcp. [Description in the php language server](https://github.com/felixfbecker/php-language-server#--tcphostport-optional)

**Comment by yyoncho** [comment]
Ok, I wanted to make sure you are aware of that option.

**Comment by kenkangxgwe** [comment]
Hi @zhougts , could you please upgrade the lsp-mode and try to register the latest php server again?
`--tcp` for `lsp-tcp-server-command` or `--tcp-server` for `lsp-tcp-connection`.

**Comment by nbfalcon** [comment]
I also found that php-ls does not work with stdio connections, only supporting *reverse* TCP. I'll make a PR to fix that one soon.

**Comment by yyoncho** [comment]
@nbfalcon you already fixed that, right?
